graylog2-web-interface/src/components/configurations/URLWhitelistInput.jsx

```diff
@@ -0,0 +1,122 @@
+  const [isWhitelisted, setIsWhitelisted] = useState(false);
+  const [currentValidationState, setCurrentValidationState] = useState(validationState);
+  const [ownValidationMessage, setOwnValidationMessage] = useState(validationMessage);
+
+  const suggestRegexWhitelistUrl = (typedUrl: string, type: string): string | Promise<any> => {
+    // eslint-disable-next-line no-template-curly-in-string
+    const keyWildcard = '${key}';
+    return type && type === 'regex' && URLUtils.isValidURL(typedUrl) ? ToolsStore.urlWhiteListGenerateRegex(typedUrl, keyWildcard) : typedUrl;
+  };
+
+  const [suggestedUrl, setSuggestedUrl] = useState(url);
+  const isError = () => currentValidationState === 'error';
+  const ref = useRef();
+
+  const handleCheckIsWhitelisted = () => {
+    const promise = ToolsStore.urlWhiteListCheck(url);
+    promise.then((result) => {
+      if (!result.is_whitelisted && validationState === null) {
+        setCurrentValidationState('error');
+        setOwnValidationMessage(`URL ${suggestedUrl} is not whitelisted or not valid URL.`);
+      } else {
+        setOwnValidationMessage(validationMessage);
+        setCurrentValidationState(validationState);
+      }
+      setIsWhitelisted(result.is_whitelisted);
+    });
+  };
+
+  const checkIsWhitelisted = () => {
+    if (url) {
+      const suggestion = suggestRegexWhitelistUrl(url, formType);
+      if (typeof suggestion === 'object') {
+        suggestion.then((result) => {
+          setSuggestedUrl(result.regex);
+          handleCheckIsWhitelisted();
+        });
+      } else {
+        setSuggestedUrl(url);
+        handleCheckIsWhitelisted();
+      }
+    }
+  };
+
+
+  const onUpdate = () => {
+    UIUtils.triggerInput(ref.current);
+    checkIsWhitelisted();
+  };
+
+  const handleChange = (event: SyntheticInputEvent<EventTarget>) => {
+    handleFormEvent(event);
+  };
+
+  useEffect(() => {
+    checkIsWhitelisted();
+  }, [url, validationState]);
+
+
+  const addButton = isError() && !isWhitelisted ? <URLWhitelistFormModal newUrlEntry={suggestedUrl} onUpdate={onUpdate} formType={formType} /> : '';
+  const helpMessage = <>{ownValidationMessage} {addButton}</>;
+  return (
+    <Input type="text"
+           id="url"
+           name="url"
+           label={label}
+           ref={ref}
+           autoFocus
+           required
+           onChange={handleChange}
+           help={helpMessage}
+           bsStyle={currentValidationState}
+           value={url}
+           labelClassName={labelClassName}
+           wrapperClassName={wrapperClassName} />
+  );
+};
+
+URLWhitelistInput.propTypes = {
+  label: PropTypes.string.isRequired,
+  handleFormEvent: PropTypes.func.isRequired,
+  validationState: PropTypes.string,
+  validationMessage: PropTypes.oneOfType([
+    PropTypes.object,
+    PropTypes.string,
+  ]),
+  url: PropTypes.string,
+  labelClassName: PropTypes.string,
+  wrapperClassName: PropTypes.string,
+  formType: PropTypes.string,
```

Comment:
I think we should restrict the actual values that are accepted for this prop. I also think that the name could be more descriptive, something like `urlType` is clearer to me than `formType`.